packages/analytics/amplify_analytics_pinpoint/ios/Classes/SwiftAmplifyAnalyticsPinpointPlugin.swift

```diff
@@ -0,0 +1,62 @@
+import Flutter
+import UIKit
+
+public class SwiftAmplifyAnalyticsPinpointPlugin: NSObject, FlutterPlugin, PigeonDataProvider {
+    public func getEndpointId(completion: @escaping (String?, FlutterError?) -> Void) {
+        let keychain = KeychainReader()
+        guard let uniqueId = keychain.getEndpointId() else {
+            completion (nil, nil)
+            return
+        }
+        completion (uniqueId, nil)
+    }
+
+    public func initializePinpointAppId(_ pinpointAppId: String, completion: @escaping (FlutterError?) -> Void) {
+        // Only needed in Android
+        completion(nil)
+    }
+
+    public static func register(with registrar: FlutterPluginRegistrar) {
+        let instance = SwiftAmplifyAnalyticsPinpointPlugin()
+        PigeonDataProviderSetup(registrar.messenger(), instance)
+    }
+}
+
+class KeychainReader {
+    
+    static let pinpointAttrService = "com.amazonaws.AWSPinpointContext"
+    static let uniqueIdKey = "com.amazonaws.AWSPinpointContextKeychainUniqueIdKey"
+    
+    func getEndpointId() -> String? {
+        return string(forKey: KeychainReader.uniqueIdKey)
+    }
+    
+    private func string(forKey key: String) -> String? {
+        guard let data = data(forKey: key) else {
+            return nil
+        }
+        return String(data: data, encoding: .utf8)
+    }
+
+    private func data(forKey key: String) -> Data? {
+        let query = [
+          kSecClass: kSecClassGenericPassword,
+          kSecAttrService: KeychainReader.pinpointAttrService,
+          kSecAttrAccount: key,
+          kSecReturnData: true
+        ] as CFDictionary
+
+        var result: AnyObject?
+        let status = SecItemCopyMatching(query, &result)
+
+        guard status != errSecItemNotFound else {
+            return nil
+        }
+
+        guard status == errSecSuccess else {
+            return nil
```

Comment:
Is this what iOS does? I thought there were errors worth surfacing here, e.g. keychain access issues due to permissions.

Reply:
I believe iOS just logs the issue, we can do the same.  I don't think we should throw an exception and disrupt code flow over an initialization issue here.

Reply:
guard status == errSecSuccess else {
            print("Read for \(key) failed")
            return nil
        }

Reply:
not relevant anymore, ios code removed